### PR TITLE
fix: taos shell support ws client

### PR DESCRIPTION
**Pull request by zhaoyanggh**

### CODE REVIEW

File: src/kit/shell/src/shellEngine.c [hunk 1]
```diff
@@ -1362,45 +1366,60 @@ int wsclient_conn() {
...
-  cJSON *res = cJSON_Parse(recv_buffer + start);
-  if (res == NULL) {
-    fprintf(stderr, "fail to parse response into json: %s\n", recv_buffer + start);
+  }
+  parser->offset = pos;
+  parser->payload_length = payload_length;
+}
+
+char *wsclient_get_response() {
+  uint8_t *recv_buffer = calloc(1, 1024);
```
Comment (sangshuduo) [inline src/kit/shell/src/shellEngine.c]: How about putting it on the stack as an array of just 1024 bytes if you don't want to check allocation succeed?

**Review by sangshuduo — COMMENTED** [review]
commented

File: src/kit/shell/CMakeLists.txt [hunk 2]
```diff
@@ -9,7 +9,8 @@ IF (TD_LINUX)
   AUX_SOURCE_DIRECTORY(./src SRC)
   LIST(REMOVE_ITEM SRC ./src/shellWindows.c)
   LIST(REMOVE_ITEM SRC ./src/shellDarwin.c)
-  ADD_EXECUTABLE(shell ${SRC})
+  set(CMAKE_C_FLAGS "-Wall -Wshadow -std=c99 -std=gnu11 -static-libasan -fsanitize=address -fsanitize=undefined -fno-sanitize-recover=all -fsanitize=float-divide-by-zero -fsanitize=float-cast-overflow -fno-sanitize=null -fno-sanitize=alignment -O0 -g3 -DDEBUG0ww0")
```
Comment (sangshuduo) [inline src/kit/shell/CMakeLists.txt]: I think no need use sanitizer by default.

File: src/kit/shell/src/shellEngine.c [hunk 3]
```diff
@@ -1362,45 +1368,71 @@ int wsclient_conn() {
...
   }
-  return res;
+  wsclient_parse_frame(&parser, recv_buffer);
+  if (parser.frame == PING_FRAME) {
+    if (wsclient_send("pong", PONG_FRAME)) {
+      return NULL;
+    }
+    return wsclient_get_response();
+  }
+  char* response = calloc(1, parser.payload_length + 1);
```
Comment (sangshuduo) [inline src/kit/shell/src/shellEngine.c:1424]: here should check if calloc() success

**Review by sangshuduo — APPROVED** [review]
LGTM with comments.

**Review by afwerar — COMMENTED** [review]
LGTM

**Pull request merged**